pkg/kv/kvserver/replica_proposal_buf.go

```diff
@@ -698,11 +704,21 @@ func (b *propBuf) maybeRejectUnsafeProposalLocked(
 		if li.iAmTheLeader {
 			return false
 		}
-		leaderKnownAndEligible := li.leaderKnown && li.leaderEligibleForLease
+
+		reject := false
+		if !li.leaderKnown() && RejectLeaseOnLeaderUnknown.Get(&b.settings.SV) {
```

Comment:
Do we want this to take `&& !ownsCurrentLease` too?

Reply:
We probably do. I think we hit this path for lease extensions too, and it seems prudent to allow the current leaseholder to attempt a lease extension even if there isn't a current leader, in case the range recovers soon.

Reply:
Done. Moved this case up (see second commit). All the code here and below now handles "I am not leader" && "I don't know the lease".